### PR TITLE
calc: add support for .uno:ConvertFormulaToValue command

### PR DESCRIPTION
**Pull request by rparth07**
Change-Id: Ia2f48eca778c396ae1fa17afe2d92d9631aadad5

* Resolves: #7984
* Target version: master 

### Summary
- Added support to replace a formula with its value in Data > Formula to Value
- Here, is the LOKit patch for the same: https://gerrit.libreoffice.org/c/core/+/168387

### Checklist

- [X] Code is properly formatted
- [X] All commits have Change-Id
- [ ] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [X] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by hfiguiere** [comment]
shouldn't there be LOKit patch to enable the command status?

**Comment by rparth07** [comment]
> shouldn't there be LOKit patch to enable the command status?

Do you have any reference on how to enable command status using LOKit patch? I'm not much aware about it.

**Comment by rparth07** [comment]
> > shouldn't there be LOKit patch to enable the command status?
> 
> Do you have any reference on how to enable command status using LOKit patch? I'm not much aware about it.

After conducting some research, I discovered the following approach: 
https://gerrit.libreoffice.org/c/core/+/132585

To enable the command, it appears that I need to modify two files in the core:
1. desktop/source/lib/init.cxx
2. sfx2/source/control/unoctitm.cxx

Could you please confirm if this is the correct way?

File: browser/src/control/Control.NotebookbarCalc.js [hunk 1]
```diff
@@ -1735,11 +1735,35 @@ L.Control.NotebookbarCalc = L.Control.NotebookbarWriter.extend({
 				'vertical': 'true'
 			},
 			{
-				'id': 'data-calculate',
```
Comment (codewithvk) [inline browser/src/control/Control.NotebookbarCalc.js:1738]: @pedropintosilva Are we really using `Recalculate` in Online? I think it is useful once we have a check menu of `AutoCalculate`.

File: browser/src/control/Control.NotebookbarCalc.js [hunk 2]
```diff
@@ -1735,11 +1735,35 @@ L.Control.NotebookbarCalc = L.Control.NotebookbarWriter.extend({
...
+								'type': 'toolitem',
+								'text': _UNO('.uno:ConvertFormulaToValue', 'spreadsheet'),
+								'command': '.uno:ConvertFormulaToValue',
+								'image': false,
+								'accessibility': { focusBack: true,	combination: 'R', de: null }
+							}
+						]
+					}
+				],
+				'vertical': 'true'
```
Comment (codewithvk) [inline browser/src/control/Control.NotebookbarCalc.js:1766]: I would prefer to have a submenu for Recalculate and Formula to Value under the Calculate menu, just like we have in LO, but I would love to hear feedback from @pedropintosilva .  

Screenshot from 2024-06-01 13-43-23

**Comment by rparth07** [comment]
@eszkadev and @pedropintosilva would love to hear your thoughs on this.

**Comment by rparth07** [comment]
here, is the LOKit patch to enable the uno command: https://gerrit.libreoffice.org/c/core/+/168387

**Comment by Darshan-upadhyay1110** [comment]
hello @rparth07 i just encounter the same thing you are having.And thanks to @eszkadev i solved it :)

Follow these steps.

Remove your hard coded change in unocommand.js
then run `    scripts/unocommands.py --update /path/to/online /path/to/loffice `

This will simply fix your issue.

- Please close your CORE pr as it will be not need now after above steps.

Thanks 🙇🏼

**Review by pedropintosilva — APPROVED** [review]
Thanks!

I think we should merge this , it just needs an icon.

Later we can think if we should have a dropdown or now

**Comment by Darshan-upadhyay1110** [comment]
@rparth07 Please rebase this PR. 
CC: @pedropintosilva

**Comment by rparth07** [comment]
> @rparth07 Please rebase this PR. CC: @pedropintosilva

Rebased!

**Pull request merged**